### PR TITLE
Changing JSON function to get all dependencies

### PR DESCRIPTION
**Pull request by alex-page**
Previously the function only got the direct dependencies for a component. We now need it to find all dependencies in the trail.

### CODE REVIEW

File: scripts/helper.js [hunk 1]
```diff
@@ -637,13 +637,12 @@ HELPER.generate = (() => {
 			// iterate over all packages
 			if( allModules !== undefined && allModules.length > 0 ) {
 				for( let module of allModules ) {
-					const pkgPath = Path.normalize(`${ packagesPath }/${ module }/package.json`);
-					packageJson = require( pkgPath ); // read the package.json
+					const packageJson = require( Path.normalize( `${ packagesPath }/${ module }/package.json` ) );
```
Comment (dominikwilkowski) [inline scripts/helper.js]: `packageJson` is being initialized at line 634...

File: scripts/helper.js [hunk 2]
```diff
@@ -660,6 +659,39 @@ HELPER.generate = (() => {
 		},
 
 		/**
+		 * Get all the dependencies and their child dependencies
+		 *
+		 * @param {object} dependencies      - An object containing dependency { name: version }
```
Comment (dominikwilkowski) [inline scripts/helper.js]: A good way to describe the shape of an option in jsdocs is to add the keys like so:
```js
		/**
		 * Get all the dependencies and their child dependencies
		 *
		 * @param {object} dependencies         - An object containing dependency
		 * @param {object} dependencies.[name]  - The version string with the name as the key
		 * @param {object} dependencyBundle     - An empty object to add the found dependencies to
		 *
		 * @return {array} dependencyBundle  - An object containing all of the dependencies found
		 */
```

File: scripts/helper.js (hunk 1 above)
Comment (alex-page) [reply scripts/helper.js]: Yeah nice, missed that!

File: scripts/helper.js (hunk 2 above)
Comment (alex-page) [reply scripts/helper.js]: Will add that now!

**Pull request merged**